tests/e2e/bulker.test.ts

```diff
@@ -0,0 +1,1026 @@
+import { expect } from "chai";
+import { utils, constants } from "ethers";
+import hre, { ethers } from "hardhat";
+import { deal } from "hardhat-deal";
+
+import { BaseProvider } from "@ethersproject/providers";
+import PercentMath from "@morpho-labs/ethers-utils/lib/maths/PercentMath";
+import {
+  ERC20__factory,
+  Weth__factory,
+  ERC20,
+  Weth,
+  MorphoAaveV3,
+  MorphoAaveV3__factory,
+  StEth__factory,
+  StEth,
+  WstETH,
+  WstETH__factory,
+} from "@morpho-labs/morpho-ethers-contract";
+import {
+  time,
+  SnapshotRestorer,
+} from "@nomicfoundation/hardhat-network-helpers";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+
+import { MorphoAaveV3Adapter } from "../../src";
+import CONTRACT_ADDRESSES from "../../src/contracts/addresses";
+import addresses from "../../src/contracts/addresses";
+import { Underlying } from "../../src/mocks/markets";
+import BulkerTxHandler from "../../src/txHandler/Bulker.TxHandler";
+import { MaxCapacityLimiter, TransactionType } from "../../src/types";
+import { delay } from "../../src/utils/promises";
+import { approxEqual } from "../helpers/bn";
+
+const ONE_ETH = utils.parseEther("1");
+const dust = utils.parseEther("0.000001");
+const interests = utils.parseEther("0.0001");
+
+describe("MorphoAaveV3 Bulker", () => {
+  let snapshot: SnapshotRestorer;
+  let initialBlock: number;
+  let morphoUser: SignerWithAddress;
+  let morphoAdapter: MorphoAaveV3Adapter;
+  let morphoAaveV3: MorphoAaveV3;
+  let owner: string;
+  let weth: Weth;
+  let steth: StEth;
+  let dai: ERC20;
+  let wsteth: WstETH;
+  const initialWethBalance = utils.parseEther("5");
+  const initialDaiBalance = utils.parseEther("5000");
+  const initialStEthBalance = utils.parseEther("5");
+  let bulker: BulkerTxHandler;
+
+  /** Approve tokens spending for Bulker.
+   * We unfortunately can't rely on beforeEach because we need to test two
+   * different cases which are mutually exclusive.
+   * We need to test with a Bulker approval and with a Permit2 approval.
+   * The idea of the function is to allow the Bulker to be easily tested with both.
+   */
```

Comment:
If both situations are mutually exclusive, it should be tested in two different test suite

Reply:
Not 100% categoric on this point, must be discussed